src/compiler/interface.jl

```diff
@@ -11,6 +11,27 @@ Context() = Context(nothing)
 
 cache(cx::Context) = cx.cache === nothing ? (cx.cache = IdDict()) : cx.cache
 
+"""
+  NoContext <: Zygote.AContext
+
+The `NoContext` context assumes that there is no global context, and errors if any part of a
+function that is being differentiated interacts with either globals or tasks.
+
+The benefit of using such a construct is the elimination of overhead associated with the
+`cache` `IdDict` in a standard `Context`, which can be prohibitively large in certain
+settings.
+"""
+struct NoContext <: Zygote.AContext end
+
+# There is no cache, so there are never any fields.
+cache(cx::NoContext) = (cache_fields=nothing)
+
+# The is no cache, so this context never has any fields.
+Base.haskey(cx::NoContext, x) = false
```

Comment:
Should this be `false` or should it error?
I guess `false` is safe, because it would have already errored when it tried to set a value in the cache?